watertap/core/wt_database.py

```diff
@@ -192,23 +190,25 @@ def _return_component_list(self):
         return self._component_list
 
     def _get_technology(self, technology):
+        # If data is already in cached files, then return that data
         if technology in self._cached_files:
-            # If data is already in cached files, return
             return self._cached_files[technology]
-        else:
-            # Else load data from required file
-            try:
-                with open(os.path.join(self._dbpath, technology + ".yaml"), "r") as f:
-                    lines = f.read()
-                    f.close()
-            except OSError:
-                raise KeyError(f"Could not find entry for {technology} in database.")
 
-            fdata = yaml.load(lines, yaml.Loader)
+        # If user provided a YAML file, use the filename as the technology
+        if hasattr(self, "_filename"):
+            technology = self._filename
```

Comment:
Shouldn't we have a test that includes a Database instance along with a unit model that makes use of this user-provided file path? I would want to see if anything breaks when using the "typical" sequence of methods (e.g., `load_parameters_from_database`, `get_unit_operation_parameters`) that use in our tests for ZO unit models and/or flowsheets.